shared/reducers/settings.js

```diff
@@ -1,22 +1,21 @@
 // @flow
 import logger from '../logger'
+import * as I from 'immutable'
 import * as SettingsGen from '../actions/settings-gen'
 import * as Types from '../constants/types/settings'
 import * as Constants from '../constants/settings'
 import * as Flow from '../util/flow'
 
-function reducer(state: Types.State = Constants.initialState, action: SettingsGen.Actions): Types.State {
+const initialState: Types.State = Constants.makeState()
+
+function reducer(state: Types.State = initialState, action: SettingsGen.Actions): Types.State {
   switch (action.type) {
     case SettingsGen.resetStore:
-      return {...Constants.initialState}
+      return initialState
     case SettingsGen.setAllowDeleteAccount:
-      const {allow} = action.payload
-      return {
-        ...Constants.initialState,
```

Comment:
This was a bug -- going to delete me (but not yet accepting it) wiped out your settings store because the spread started with initialState instead of current state.